app/src/main/java/com/arctouch/giphy/presentation/main/MainActivity.kt

```diff
@@ -0,0 +1,32 @@
+package com.arctouch.giphy.presentation.main
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import com.arctouch.giphy.R
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var viewModel: MainViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        setupViewModel()
+        getDataButton.setOnClickListener {
```

Comment:
You could create a function called `setupViews` or something related to remove this kind of code from onCreate.